graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphQLServiceBuilder.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.graphql;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+import org.dataloader.DataLoaderRegistry;
+
+import com.google.common.collect.ImmutableList;
+
+import graphql.GraphQL;
+import graphql.execution.instrumentation.ChainedInstrumentation;
+import graphql.execution.instrumentation.Instrumentation;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.GraphQLTypeVisitor;
+import graphql.schema.SchemaTransformer;
+import graphql.schema.idl.RuntimeWiring;
+import graphql.schema.idl.RuntimeWiring.Builder;
+import graphql.schema.idl.SchemaGenerator;
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+
+/**
+ * Constructs a {@link GraphQLService} to serve GraphQL services from within Armeria.
+ */
+final class GraphQLServiceBuilder {
+
+    private final ImmutableList.Builder<File> schemaFileBuilder = ImmutableList.builder();
+
+    private final ImmutableList.Builder<RuntimeWiringConfigurator> runtimeWiringConfiguratorBuilder =
+                                                                                        ImmutableList.builder();
+    private final ImmutableList.Builder<GraphQLTypeVisitor> typeVisitorBuilder = ImmutableList.builder();
+    private final ImmutableList.Builder<Instrumentation> instrumentationBuilder = ImmutableList.builder();
+    private final ImmutableList.Builder<Consumer<GraphQL.Builder>> configurerBuilder = ImmutableList.builder();
+
+    @Nullable
+    private DataLoaderRegistry dataLoaderRegistry;
+
+    GraphQLServiceBuilder() {}
+
+    /**
+     * Sets the schema file.
+     * If not set, the `schema.graphql` or `schema.graphqls` will be imported from the resource.
+     */
+    public GraphQLServiceBuilder schemaFile(File... schemaFile) {
+        return schemaFile(ImmutableList.copyOf(requireNonNull(schemaFile, "schemaFile")));
+    }
+
+    /**
+     * Sets the schema file.
```

Comment:
Global comment: Please use `Adds` if the specified elements are accumulated. :-)
```suggestion
     * Adds the schema {@link File}.
```